# Naive Bayes

### Problem Statement

    - Preparing classifiction model for Salary Data

In [78]:
import warnings
warnings.filterwarnings('ignore')

### Loaing Train Data

In [128]:
import pandas as pd
import numpy as np

train = pd.read_csv("~/desktop/Digi 360/Module 22/Datasets-6/SalaryData_Train.csv", encoding = 'latin1')
train.head()

age          workclass   education  educationno        maritalstatus  \
0   39          State-gov   Bachelors           13        Never-married   
1   50   Self-emp-not-inc   Bachelors           13   Married-civ-spouse   
2   38            Private     HS-grad            9             Divorced   
3   53            Private        11th            7   Married-civ-spouse   
4   28            Private   Bachelors           13   Married-civ-spouse   

           occupation    relationship    race      sex  capitalgain  \
0        Adm-clerical   Not-in-family   White     Male         2174   
1     Exec-managerial         Husband   White     Male            0   
2   Handlers-cleaners   Not-in-family   White     Male            0   
3   Handlers-cleaners         Husband   Black     Male            0   
4      Prof-specialty            Wife   Black   Female            0   

   capitalloss  hoursperweek          native  Salary  
0            0            40   United-States   <=50K  
1            0            13   United-States   <=50K  
2            0            40   United-States   <=50K  
3            0            40   United-States   <=50K  
4            0            40            Cuba   <=50K

In [129]:
train.Salary.value_counts()

 <=50K    22653
 >50K      7508
Name: Salary, dtype: int64

In [130]:
# mapping salary to 0 and 1
train['Salary'] = train['Salary'].apply(lambda x: 0 if x==' <=50K' else 1)
train.head()

age          workclass   education  educationno        maritalstatus  \
0   39          State-gov   Bachelors           13        Never-married   
1   50   Self-emp-not-inc   Bachelors           13   Married-civ-spouse   
2   38            Private     HS-grad            9             Divorced   
3   53            Private        11th            7   Married-civ-spouse   
4   28            Private   Bachelors           13   Married-civ-spouse   

           occupation    relationship    race      sex  capitalgain  \
0        Adm-clerical   Not-in-family   White     Male         2174   
1     Exec-managerial         Husband   White     Male            0   
2   Handlers-cleaners   Not-in-family   White     Male            0   
3   Handlers-cleaners         Husband   Black     Male            0   
4      Prof-specialty            Wife   Black   Female            0   

   capitalloss  hoursperweek          native  Salary  
0            0            40   United-States       0  
1            0            13   United-States       0  
2            0            40   United-States       0  
3            0            40   United-States       0  
4            0            40            Cuba       0

In [131]:
train.Salary.value_counts()

0    22653
1     7508
Name: Salary, dtype: int64

### Encoding the categorical varibales in Train data

In [132]:
dummies = pd.get_dummies(train[['workclass','education','maritalstatus','occupation','relationship','race', 'sex','native']],drop_first=True)
dummies.head()

workclass_ Local-gov  workclass_ Private  workclass_ Self-emp-inc  \
0                     0                   0                        0   
1                     0                   0                        0   
2                     0                   1                        0   
3                     0                   1                        0   
4                     0                   1                        0   

   workclass_ Self-emp-not-inc  workclass_ State-gov  workclass_ Without-pay  \
0                            0                     1                       0   
1                            1                     0                       0   
2                            0                     0                       0   
3                            0                     0                       0   
4                            0                     0                       0   

   education_ 11th  education_ 12th  education_ 1st-4th  education_ 5th-6th  \
0                0                0                   0                   0   
1                0                0                   0                   0   
2                0                0                   0                   0   
3                1                0                   0                   0   
4                0                0                   0                   0   

   ...  native_ Portugal  native_ Puerto-Rico  native_ Scotland  \
0  ...                 0                    0                 0   
1  ...                 0                    0                 0   
2  ...                 0                    0                 0   
3  ...                 0                    0                 0   
4  ...                 0                    0                 0   

   native_ South  native_ Taiwan  native_ Thailand  native_ Trinadad&Tobago  \
0              0               0                 0                        0   
1              0               0                 0                        0   
2              0               0                 0                        0   
3              0               0                 0                        0   
4              0               0                 0                        0   

   native_ United-States  native_ Vietnam  native_ Yugoslavia  
0                      1                0                   0  
1                      1                0                   0  
2                      1                0                   0  
3                      1                0                   0  
4                      0                0                   0  

[5 rows x 89 columns]

In [133]:
#Let's concat dummies dataframe with original dataframe and drop the original columns

train = pd.concat([train, dummies], axis = 1)
train.head()

age          workclass   education  educationno        maritalstatus  \
0   39          State-gov   Bachelors           13        Never-married   
1   50   Self-emp-not-inc   Bachelors           13   Married-civ-spouse   
2   38            Private     HS-grad            9             Divorced   
3   53            Private        11th            7   Married-civ-spouse   
4   28            Private   Bachelors           13   Married-civ-spouse   

           occupation    relationship    race      sex  capitalgain  ...  \
0        Adm-clerical   Not-in-family   White     Male         2174  ...   
1     Exec-managerial         Husband   White     Male            0  ...   
2   Handlers-cleaners   Not-in-family   White     Male            0  ...   
3   Handlers-cleaners         Husband   Black     Male            0  ...   
4      Prof-specialty            Wife   Black   Female            0  ...   

   native_ Portugal  native_ Puerto-Rico native_ Scotland  native_ South  \
0                 0                    0                0              0   
1                 0                    0                0              0   
2                 0                    0                0              0   
3                 0                    0                0              0   
4                 0                    0                0              0   

   native_ Taiwan  native_ Thailand  native_ Trinadad&Tobago  \
0               0                 0                        0   
1               0                 0                        0   
2               0                 0                        0   
3               0                 0                        0   
4               0                 0                        0   

   native_ United-States  native_ Vietnam  native_ Yugoslavia  
0                      1                0                   0  
1                      1                0                   0  
2                      1                0                   0  
3                      1                0                   0  
4                      0                0                   0  

[5 rows x 103 columns]

In [134]:
# Let's drop the original columns since they are already converted to numeric values.
train.drop(['workclass','education','maritalstatus','occupation','relationship','race', 'sex','native'], axis = 1, inplace = True)
train.head()

age  educationno  capitalgain  capitalloss  hoursperweek  Salary  \
0   39           13         2174            0            40       0   
1   50           13            0            0            13       0   
2   38            9            0            0            40       0   
3   53            7            0            0            40       0   
4   28           13            0            0            40       0   

   workclass_ Local-gov  workclass_ Private  workclass_ Self-emp-inc  \
0                     0                   0                        0   
1                     0                   0                        0   
2                     0                   1                        0   
3                     0                   1                        0   
4                     0                   1                        0   

   workclass_ Self-emp-not-inc  ...  native_ Portugal  native_ Puerto-Rico  \
0                            0  ...                 0                    0   
1                            1  ...                 0                    0   
2                            0  ...                 0                    0   
3                            0  ...                 0                    0   
4                            0  ...                 0                    0   

   native_ Scotland  native_ South  native_ Taiwan  native_ Thailand  \
0                 0              0               0                 0   
1                 0              0               0                 0   
2                 0              0               0                 0   
3                 0              0               0                 0   
4                 0              0               0                 0   

   native_ Trinadad&Tobago  native_ United-States  native_ Vietnam  \
0                        0                      1                0   
1                        0                      1                0   
2                        0                      1                0   
3                        0                      1                0   
4                        0                      0                0   

   native_ Yugoslavia  
0                   0  
1                   0  
2                   0  
3                   0  
4                   0  

[5 rows x 95 columns]

In [135]:
X_train = train.drop(['Salary'], axis = 1)
X_train.head()

age  educationno  capitalgain  capitalloss  hoursperweek  \
0   39           13         2174            0            40   
1   50           13            0            0            13   
2   38            9            0            0            40   
3   53            7            0            0            40   
4   28           13            0            0            40   

   workclass_ Local-gov  workclass_ Private  workclass_ Self-emp-inc  \
0                     0                   0                        0   
1                     0                   0                        0   
2                     0                   1                        0   
3                     0                   1                        0   
4                     0                   1                        0   

   workclass_ Self-emp-not-inc  workclass_ State-gov  ...  native_ Portugal  \
0                            0                     1  ...                 0   
1                            1                     0  ...                 0   
2                            0                     0  ...                 0   
3                            0                     0  ...                 0   
4                            0                     0  ...                 0   

   native_ Puerto-Rico  native_ Scotland  native_ South  native_ Taiwan  \
0                    0                 0              0               0   
1                    0                 0              0               0   
2                    0                 0              0               0   
3                    0                 0              0               0   
4                    0                 0              0               0   

   native_ Thailand  native_ Trinadad&Tobago  native_ United-States  \
0                 0                        0                      1   
1                 0                        0                      1   
2                 0                        0                      1   
3                 0                        0                      1   
4                 0                        0                      0   

   native_ Vietnam  native_ Yugoslavia  
0                0                   0  
1                0                   0  
2                0                   0  
3                0                   0  
4                0                   0  

[5 rows x 94 columns]

In [136]:
y_train = train.Salary
y_train.head()

0    0
1    0
2    0
3    0
4    0
Name: Salary, dtype: int64

In [139]:
y_train.value_counts()

0    22653
1     7508
Name: Salary, dtype: int64

### Loading Test Data

In [148]:
test = pd.read_csv("~/desktop/Digi 360/Module 22/Datasets-6/SalaryData_Test.csv", encoding = 'latin1')
test.head()

age   workclass      education  educationno        maritalstatus  \
0   25     Private           11th            7        Never-married   
1   38     Private        HS-grad            9   Married-civ-spouse   
2   28   Local-gov     Assoc-acdm           12   Married-civ-spouse   
3   44     Private   Some-college           10   Married-civ-spouse   
4   34     Private           10th            6        Never-married   

           occupation    relationship    race    sex  capitalgain  \
0   Machine-op-inspct       Own-child   Black   Male            0   
1     Farming-fishing         Husband   White   Male            0   
2     Protective-serv         Husband   White   Male            0   
3   Machine-op-inspct         Husband   Black   Male         7688   
4       Other-service   Not-in-family   White   Male            0   

   capitalloss  hoursperweek          native  Salary  
0            0            40   United-States   <=50K  
1            0            50   United-States   <=50K  
2            0            40   United-States    >50K  
3            0            40   United-States    >50K  
4            0            30   United-States   <=50K

In [149]:
test.Salary.value_counts()

 <=50K    11360
 >50K      3700
Name: Salary, dtype: int64

In [150]:
# mapping salary to 0 and 1
test['Salary'] = test['Salary'].apply(lambda x: 0 if x==' <=50K' else 1)
test.head()

age   workclass      education  educationno        maritalstatus  \
0   25     Private           11th            7        Never-married   
1   38     Private        HS-grad            9   Married-civ-spouse   
2   28   Local-gov     Assoc-acdm           12   Married-civ-spouse   
3   44     Private   Some-college           10   Married-civ-spouse   
4   34     Private           10th            6        Never-married   

           occupation    relationship    race    sex  capitalgain  \
0   Machine-op-inspct       Own-child   Black   Male            0   
1     Farming-fishing         Husband   White   Male            0   
2     Protective-serv         Husband   White   Male            0   
3   Machine-op-inspct         Husband   Black   Male         7688   
4       Other-service   Not-in-family   White   Male            0   

   capitalloss  hoursperweek          native  Salary  
0            0            40   United-States       0  
1            0            50   United-States       0  
2            0            40   United-States       1  
3            0            40   United-States       1  
4            0            30   United-States       0

In [151]:
test.Salary.value_counts()

0    11360
1     3700
Name: Salary, dtype: int64

### Encoding the categorical varibales in Test data

In [152]:
dummies = pd.get_dummies(test[['workclass','education','maritalstatus','occupation','relationship','race', 'sex','native']],drop_first=True)
dummies.head()

workclass_ Local-gov  workclass_ Private  workclass_ Self-emp-inc  \
0                     0                   1                        0   
1                     0                   1                        0   
2                     1                   0                        0   
3                     0                   1                        0   
4                     0                   1                        0   

   workclass_ Self-emp-not-inc  workclass_ State-gov  workclass_ Without-pay  \
0                            0                     0                       0   
1                            0                     0                       0   
2                            0                     0                       0   
3                            0                     0                       0   
4                            0                     0                       0   

   education_ 11th  education_ 12th  education_ 1st-4th  education_ 5th-6th  \
0                1                0                   0                   0   
1                0                0                   0                   0   
2                0                0                   0                   0   
3                0                0                   0                   0   
4                0                0                   0                   0   

   ...  native_ Portugal  native_ Puerto-Rico  native_ Scotland  \
0  ...                 0                    0                 0   
1  ...                 0                    0                 0   
2  ...                 0                    0                 0   
3  ...                 0                    0                 0   
4  ...                 0                    0                 0   

   native_ South  native_ Taiwan  native_ Thailand  native_ Trinadad&Tobago  \
0              0               0                 0                        0   
1              0               0                 0                        0   
2              0               0                 0                        0   
3              0               0                 0                        0   
4              0               0                 0                        0   

   native_ United-States  native_ Vietnam  native_ Yugoslavia  
0                      1                0                   0  
1                      1                0                   0  
2                      1                0                   0  
3                      1                0                   0  
4                      1                0                   0  

[5 rows x 89 columns]

In [153]:
#Let's concat dummies dataframe with original dataframe and drop the original columns

test = pd.concat([test, dummies], axis = 1)
test.head()

age   workclass      education  educationno        maritalstatus  \
0   25     Private           11th            7        Never-married   
1   38     Private        HS-grad            9   Married-civ-spouse   
2   28   Local-gov     Assoc-acdm           12   Married-civ-spouse   
3   44     Private   Some-college           10   Married-civ-spouse   
4   34     Private           10th            6        Never-married   

           occupation    relationship    race    sex  capitalgain  ...  \
0   Machine-op-inspct       Own-child   Black   Male            0  ...   
1     Farming-fishing         Husband   White   Male            0  ...   
2     Protective-serv         Husband   White   Male            0  ...   
3   Machine-op-inspct         Husband   Black   Male         7688  ...   
4       Other-service   Not-in-family   White   Male            0  ...   

   native_ Portugal  native_ Puerto-Rico native_ Scotland  native_ South  \
0                 0                    0                0              0   
1                 0                    0                0              0   
2                 0                    0                0              0   
3                 0                    0                0              0   
4                 0                    0                0              0   

   native_ Taiwan  native_ Thailand  native_ Trinadad&Tobago  \
0               0                 0                        0   
1               0                 0                        0   
2               0                 0                        0   
3               0                 0                        0   
4               0                 0                        0   

   native_ United-States  native_ Vietnam  native_ Yugoslavia  
0                      1                0                   0  
1                      1                0                   0  
2                      1                0                   0  
3                      1                0                   0  
4                      1                0                   0  

[5 rows x 103 columns]

In [154]:
# Let's drop the original columns since they are already converted to numeric values.
test.drop(['workclass','education','maritalstatus','occupation','relationship','race', 'sex','native'], axis = 1, inplace = True)
test.head()

age  educationno  capitalgain  capitalloss  hoursperweek  Salary  \
0   25            7            0            0            40       0   
1   38            9            0            0            50       0   
2   28           12            0            0            40       1   
3   44           10         7688            0            40       1   
4   34            6            0            0            30       0   

   workclass_ Local-gov  workclass_ Private  workclass_ Self-emp-inc  \
0                     0                   1                        0   
1                     0                   1                        0   
2                     1                   0                        0   
3                     0                   1                        0   
4                     0                   1                        0   

   workclass_ Self-emp-not-inc  ...  native_ Portugal  native_ Puerto-Rico  \
0                            0  ...                 0                    0   
1                            0  ...                 0                    0   
2                            0  ...                 0                    0   
3                            0  ...                 0                    0   
4                            0  ...                 0                    0   

   native_ Scotland  native_ South  native_ Taiwan  native_ Thailand  \
0                 0              0               0                 0   
1                 0              0               0                 0   
2                 0              0               0                 0   
3                 0              0               0                 0   
4                 0              0               0                 0   

   native_ Trinadad&Tobago  native_ United-States  native_ Vietnam  \
0                        0                      1                0   
1                        0                      1                0   
2                        0                      1                0   
3                        0                      1                0   
4                        0                      1                0   

   native_ Yugoslavia  
0                   0  
1                   0  
2                   0  
3                   0  
4                   0  

[5 rows x 95 columns]

In [155]:
X_test = test.drop(['Salary'], axis = 1)
X_test.head()

age  educationno  capitalgain  capitalloss  hoursperweek  \
0   25            7            0            0            40   
1   38            9            0            0            50   
2   28           12            0            0            40   
3   44           10         7688            0            40   
4   34            6            0            0            30   

   workclass_ Local-gov  workclass_ Private  workclass_ Self-emp-inc  \
0                     0                   1                        0   
1                     0                   1                        0   
2                     1                   0                        0   
3                     0                   1                        0   
4                     0                   1                        0   

   workclass_ Self-emp-not-inc  workclass_ State-gov  ...  native_ Portugal  \
0                            0                     0  ...                 0   
1                            0                     0  ...                 0   
2                            0                     0  ...                 0   
3                            0                     0  ...                 0   
4                            0                     0  ...                 0   

   native_ Puerto-Rico  native_ Scotland  native_ South  native_ Taiwan  \
0                    0                 0              0               0   
1                    0                 0              0               0   
2                    0                 0              0               0   
3                    0                 0              0               0   
4                    0                 0              0               0   

   native_ Thailand  native_ Trinadad&Tobago  native_ United-States  \
0                 0                        0                      1   
1                 0                        0                      1   
2                 0                        0                      1   
3                 0                        0                      1   
4                 0                        0                      1   

   native_ Vietnam  native_ Yugoslavia  
0                0                   0  
1                0                   0  
2                0                   0  
3                0                   0  
4                0                   0  

[5 rows x 94 columns]

In [156]:
y_test = test.Salary
y_test.head()

0    0
1    0
2    1
3    1
4    0
Name: Salary, dtype: int64

In [157]:
y_test.value_counts()

0    11360
1     3700
Name: Salary, dtype: int64

### Scaling training data

In [158]:
from sklearn.preprocessing import MinMaxScaler

scaler = MinMaxScaler()
#Let's apply MinMaxScaler to the columns except the dummy varibales
num_vars = ['age', 'educationno', 'capitalgain', 'capitalloss','hoursperweek']

train[num_vars] = scaler.fit_transform(train[num_vars])
train.describe()

age   educationno   capitalgain   capitalloss  hoursperweek  \
count  30161.000000  30161.000000  30161.000000  30161.000000  30161.000000   
mean       0.293673      0.608088      0.010921      0.020271      0.407462   
std        0.179929      0.170002      0.074065      0.092773      0.122247   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%        0.150685      0.533333      0.000000      0.000000      0.397959   
50%        0.273973      0.600000      0.000000      0.000000      0.397959   
75%        0.410959      0.800000      0.000000      0.000000      0.448980   
max        1.000000      1.000000      1.000000      1.000000      1.000000   

             Salary  workclass_ Local-gov  workclass_ Private  \
count  30161.000000          30161.000000        30161.000000   
mean       0.248931              0.068532            0.738868   
std        0.432401              0.252661            0.439259   
min        0.000000              0.000000            0.000000   
25%        0.000000              0.000000            0.000000   
50%        0.000000              0.000000            1.000000   
75%        0.000000              0.000000            1.000000   
max        1.000000              1.000000            1.000000   

       workclass_ Self-emp-inc  workclass_ Self-emp-not-inc  ...  \
count             30161.000000                 30161.000000  ...   
mean                  0.035609                     0.082855  ...   
std                   0.185316                     0.275668  ...   
min                   0.000000                     0.000000  ...   
25%                   0.000000                     0.000000  ...   
50%                   0.000000                     0.000000  ...   
75%                   0.000000                     0.000000  ...   
max                   1.000000                     1.000000  ...   

       native_ Portugal  native_ Puerto-Rico  native_ Scotland  native_ South  \
count      30161.000000         30161.000000      30161.000000   30161.000000   
mean           0.001127             0.003614          0.000365       0.002354   
std            0.033557             0.060008          0.019094       0.048462   
min            0.000000             0.000000          0.000000       0.000000   
25%            0.000000             0.000000          0.000000       0.000000   
50%            0.000000             0.000000          0.000000       0.000000   
75%            0.000000             0.000000          0.000000       0.000000   
max            1.000000             1.000000          1.000000       1.000000   

       native_ Taiwan  native_ Thailand  native_ Trinadad&Tobago  \
count    30161.000000      30161.000000             30161.000000   
mean         0.001393          0.000564                 0.000597   
std          0.037291          0.023735                 0.024423   
min          0.000000          0.000000                 0.000000   
25%          0.000000          0.000000                 0.000000   
50%          0.000000          0.000000                 0.000000   
75%          0.000000          0.000000                 0.000000   
max          1.000000          1.000000                 1.000000   

       native_ United-States  native_ Vietnam  native_ Yugoslavia  
count           30161.000000     30161.000000        30161.000000  
mean                0.911906         0.002122            0.000530  
std                 0.283436         0.046016            0.023027  
min                 0.000000         0.000000            0.000000  
25%                 1.000000         0.000000            0.000000  
50%                 1.000000         0.000000            0.000000  
75%                 1.000000         0.000000            0.000000  
max                 1.000000         1.000000            1.000000  

[8 rows x 95 columns]

### Scaling Test Data

In [159]:
test[num_vars] = scaler.fit_transform(test[num_vars])
test.describe()

age   educationno   capitalgain   capitalloss  hoursperweek  \
count  15060.000000  15060.000000  15060.000000  15060.000000  15060.000000   
mean       0.298196      0.607517      0.011203      0.023619      0.407669   
std        0.183297      0.170582      0.077033      0.107767      0.123090   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%        0.150685      0.533333      0.000000      0.000000      0.397959   
50%        0.273973      0.600000      0.000000      0.000000      0.397959   
75%        0.424658      0.800000      0.000000      0.000000      0.448980   
max        1.000000      1.000000      1.000000      1.000000      1.000000   

             Salary  workclass_ Local-gov  workclass_ Private  \
count  15060.000000          15060.000000        15060.000000   
mean       0.245684              0.068592            0.731806   
std        0.430506              0.252768            0.443034   
min        0.000000              0.000000            0.000000   
25%        0.000000              0.000000            0.000000   
50%        0.000000              0.000000            1.000000   
75%        0.000000              0.000000            1.000000   
max        1.000000              1.000000            1.000000   

       workclass_ Self-emp-inc  workclass_ Self-emp-not-inc  ...  \
count             15060.000000                 15060.000000  ...   
mean                  0.037981                     0.086122  ...   
std                   0.191158                     0.280554  ...   
min                   0.000000                     0.000000  ...   
25%                   0.000000                     0.000000  ...   
50%                   0.000000                     0.000000  ...   
75%                   0.000000                     0.000000  ...   
max                   1.000000                     1.000000  ...   

       native_ Portugal  native_ Puerto-Rico  native_ Scotland  native_ South  \
count      15060.000000         15060.000000      15060.000000   15060.000000   
mean           0.001859             0.004382          0.000598       0.001992   
std            0.043080             0.066057          0.024440       0.044589   
min            0.000000             0.000000          0.000000       0.000000   
25%            0.000000             0.000000          0.000000       0.000000   
50%            0.000000             0.000000          0.000000       0.000000   
75%            0.000000             0.000000          0.000000       0.000000   
max            1.000000             1.000000          1.000000       1.000000   

       native_ Taiwan  native_ Thailand  native_ Trinadad&Tobago  \
count    15060.000000      15060.000000             15060.000000   
mean         0.000863          0.000797                 0.000531   
std          0.029369          0.028218                 0.023043   
min          0.000000          0.000000                 0.000000   
25%          0.000000          0.000000                 0.000000   
50%          0.000000          0.000000                 0.000000   
75%          0.000000          0.000000                 0.000000   
max          1.000000          1.000000                 1.000000   

       native_ United-States  native_ Vietnam  native_ Yugoslavia  
count           15060.000000     15060.000000        15060.000000  
mean                0.915538         0.001262            0.000465  
std                 0.278089         0.035498            0.021555  
min                 0.000000         0.000000            0.000000  
25%                 1.000000         0.000000            0.000000  
50%                 1.000000         0.000000            0.000000  
75%                 1.000000         0.000000            0.000000  
max                 1.000000         1.000000            1.000000  

[8 rows x 95 columns]

### Building model

In [160]:
#Fitting classifier to the Training set
from sklearn.naive_bayes import GaussianNB
classifier = GaussianNB()
classifier.fit(X_train , y_train)

GaussianNB(priors=None, var_smoothing=1e-09)

In [161]:
# Predicting on test set

y_test_pred = classifier.predict(X_test)

In [162]:
y_test_pred

array([0, 0, 1, ..., 1, 1, 1])

In [163]:
y_test.value_counts()

0    11360
1     3700
Name: Salary, dtype: int64

### Model Evaluation

In [164]:
# printing the overall accuracy
from sklearn import metrics
metrics.accuracy_score(y_test, y_test_pred)

0.8219123505976096

In [165]:
metrics.confusion_matrix(y_test, y_test_pred)

array([[9685, 1675],
       [1007, 2693]])

In [166]:
# Making the Confusion Matrix
from sklearn.metrics import confusion_matrix
confusion_matrix = confusion_matrix(y_test, y_test_pred)
confusion_matrix

array([[9685, 1675],
       [1007, 2693]])

In [168]:
confusion = metrics.confusion_matrix(y_test, y_test_pred)
print(confusion)
TN = confusion[0, 0]
FP = confusion[0, 1]
FN = confusion[1, 0]
TP = confusion[1, 1]

[[9685 1675]
 [1007 2693]]


In [169]:
sensitivity = TP / float(FN + TP)
print("sensitivity",sensitivity)

sensitivity 0.7278378378378378


In [170]:
specificity = TN / float(TN + FP)
print("specificity",specificity)

specificity 0.8525528169014085


In [172]:
precision = TP / float(TP + FP)
print("precision",precision)
print(metrics.precision_score(y_test, y_test_pred))

precision 0.6165293040293041
0.6165293040293041


In [173]:
print("precision",precision)
print("PRECISION SCORE :",metrics.precision_score(y_test, y_test_pred))
print("RECALL SCORE :", metrics.recall_score(y_test, y_test_pred))
print("F1 SCORE :",metrics.f1_score(y_test, y_test_pred))

precision 0.6165293040293041
PRECISION SCORE : 0.6165293040293041
RECALL SCORE : 0.7278378378378378
F1 SCORE : 0.6675756073376302


### Conclusion

    - Our model predicted the salary with accuracy of 82%.